plugin/storage/integration/integration.go

```diff
@@ -121,6 +125,54 @@ func (s *StorageIntegration) waitForCondition(t *testing.T, predicate func(t *te
 	return predicate(t)
 }
 
+func (s *StorageIntegration) InitArchiveStorage(storageFactory storage.Factory, logger *zap.Logger) bool {
+	archiveFactory, ok := storageFactory.(storage.ArchiveFactory)
+	if !ok {
+		logger.Info("Archive storage not supported by the factory")
+		return false
+	}
+	reader, err := archiveFactory.CreateArchiveSpanReader()
+	if err != nil {
+		logger.Error("Cannot init archive storage reader", zap.Error(err))
+		return false
+	}
```

Comment:
pass `t` to this function and just do `require.NoError(t, err)`. It's ok to fail (require will panic) because if the storage implements archive factory per above, there's no reason for it to not return a reader, it's no longer a soft error.

Reply:
Done 👍